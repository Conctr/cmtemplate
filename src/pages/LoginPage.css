.google-button {
  height: 34px;
  width: 294px;
  border-width: 0;
  background: white;
  color: #737373;
  border-radius: 1px;
  white-space: nowrap;
  -webkit-box-shadow: 1px 1px 0px 1px rgba(0, 0, 0, 0.05);
          box-shadow: 1px 1px 0px 1px rgba(0, 0, 0, 0.05);
  -webkit-transition-property: background-color, -webkit-box-shadow;
  transition-property: background-color, -webkit-box-shadow;
  transition-property: background-color, box-shadow;
  transition-property: background-color, box-shadow, -webkit-box-shadow;
  -webkit-transition-duration: 150ms;
          transition-duration: 150ms;
  -webkit-transition-timing-function: ease-in-out;
          transition-timing-function: ease-in-out;
  padding: 0;
}
.google-button-2 {
        height: 34px;
        width: 294px;
        border-width: 0;
        background: #90a6ff;
        color: white;
        border-radius: 1px;
        white-space: nowrap;
        -webkit-box-shadow: 1px 1px 0px 1px rgba(0, 0, 0, 0.05);
                box-shadow: 1px 1px 0px 1px rgba(0, 0, 0, 0.05);
        -webkit-transition-property: background-color, -webkit-box-shadow;
        transition-property: background-color, -webkit-box-shadow;
        transition-property: background-color, box-shadow;
        transition-property: background-color, box-shadow, -webkit-box-shadow;
        -webkit-transition-duration: 150ms;
                transition-duration: 150ms;
        -webkit-transition-timing-function: ease-in-out;
                transition-timing-function: ease-in-out;
        padding: 0;
}
.google-button:focus, .google-button:hover, .google-button-2:focus, .google-button-2:hover {
  cursor: pointer;
  -webkit-box-shadow: 1px 4px 5px 1px rgba(0, 0, 0, 0.1);
          box-shadow: 1px 4px 5px 1px rgba(0, 0, 0, 0.1);
}
.google-button:active, .google-button-2:active {
  background-color: #e5e5e5;
  -webkit-box-shadow: none;
          box-shadow: none;
  -webkit-transition-duration: 10ms;
          transition-duration: 10ms;
}

.google-button__icon {
  display: inline-block;
  vertical-align: middle;
  margin: 8px 0 8px 8px;
  width: 18px;
  height: 18px;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}

.google-button__icon--plus {
  width: 27px;
}

.google-button__text {
  display: inline-block;
  vertical-align: middle;
  padding: 0 24px;
  font-size: 14px;
  font-weight: bold;
  font-family: 'Roboto',arial,sans-serif;
}
