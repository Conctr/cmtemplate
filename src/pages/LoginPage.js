import React from 'react'
import CircularProgress from "material-ui/CircularProgress"
import Divider from "material-ui/Divider"
import RaisedButton from "material-ui/RaisedButton"
import TextField from "../components/atoms/TextField"
import GoogleIcon from "react-icons/lib/fa/google"
import conctrLogo from "../imgs/conctr-logo.png"
import Logo from "../imgs/logoImage.jpg"
import backgroundImageFile from "../imgs/placeholder-background.svg"
import GoogleLogin from 'react-google-login'

// class LoginPage extends Component {
//   state = {
//     token: null,
//     error: null
//   }

//   // Google 0Auth Login
//   handleGoogleLoginSuccess = (status, response) => {
//     console.log('response', response)
//     const email = response.profile.email
//     const provider = response._provider
//     const accessToken = response.token.accessToken
//     if (status === "signIn") {
//       authSignIn(email, provider, accessToken)
//         .then(conctrUser => {
//           this.setState({ token: conctrUser.jwt })
//         })
//         .catch(err => {
//           const conctrError = {
//             conctrError: err.response.data.error
//           }
//           this.setState({ error: conctrError })
//         })
//     }
//     if (status === "register") {
//       authRegister(email, provider, accessToken)
//         .then(conctrUser => {
//           this.setState({ token: conctrUser.jwt })
//         })
//         .catch(err => {
//           const conctrError = {
//             conctrError: err.response.data.error
//           }
//           this.setState({ error: conctrError })
//         })
//     }
//   }
//   handleGoogleLoginFailure = (status, response) => {
//     if (status === "signIn") {
//       if (response.message) {
//         const googleError = {
//           error: response.message
//         }
//         this.setState({ error: googleError })
//       }
//     }
//     // register
//     if (status === "register") {
//       if (response.message) {
//         const googleError = {
//           error: response.message
//         }
//         this.setState({ error: googleError })
//       }
//     }
//   }

const LoginPage = ({
  GoogleLoginSuccess,
  GoogleLoginFailure,
  GoogleRegisterSuccess,
  GoogleRegisterFailure
}) => {
  return (
    <div>
        <GoogleLogin
          clientId="623084099025-ru09r0q5rhjguaj4n348umdjcllr3hrp.apps.googleusercontent.com"
          buttonText="Login With Google"
          onSuccess={(response) => {
            GoogleLoginSuccess(response, 'signIn')
          }}
          onFailure={(response) => {
            GoogleLoginFailure(response, 'signIn')
          }}
        />
        <GoogleLogin
          clientId="623084099025-ru09r0q5rhjguaj4n348umdjcllr3hrp.apps.googleusercontent.com"
          buttonText="Register With Google"
          onSuccess={(response) => {
            GoogleRegisterSuccess(response, 'register')
          }}
          onFailure={(response) => {
            GoogleRegisterSuccess(response, 'register')
          }}
        />
      {/* <SocialButton
        provider="google"
        appId="623084099025-ru09r0q5rhjguaj4n348umdjcllr3hrp.apps.googleusercontent.com"
        onLoginSuccess={(response) => {
          GoogleLoginSuccess(response, 'signIn')
        }}
        onLoginFailure={(response) => {
          GoogleLoginFailure(response, 'signIn')
        }}
      >
        Login with google
      </SocialButton>
      <SocialButton
        provider="google"
        appId="623084099025-ru09r0q5rhjguaj4n348umdjcllr3hrp.apps.googleusercontent.com"
        onLoginSuccess={(response) => {
          GoogleLoginSuccess(response, 'register')
        }}
        onLoginFailure={(response) => {
          GoogleLoginFailure(response, 'register')
        }}
      >
        Register With Google
      </SocialButton> */}
    </div>
  )
}

export default LoginPage


// export default class LoginPage extends Component {
//   state = {
//       createAccount: false,
//       email: "",
//       password: "",
//       loading: false
//   }

//   componentDidMount() {
//     // init google auth
//     oauthApi.start(
//       this.changeLoading,
//       this.props.setToken,
//       this.props.handleErrors
//     );
//   }

//   render() {
//     const useColor = false;
//     const backgroundColor = "#C8C8C8";

//     return (
//       <div
//         className="login-background"
//         style={
//           !useColor
//             ? {
//                 backgroundImage: `url(${backgroundImageFile})`
//               }
//             : {
//               background: `${backgroundColor}`
//             }
//         }
//       >
//         <div className="conctr-footer-box">
//           <div className="conctr-footer-text">powered by</div>
//           <a href="https://conctr.com/">
//             <img
//               src={conctrLogo}
//               alt="conctr logo"
//               className="conctr-footer-logo"
//             />
//           </a>
//         </div>
//         <div className="welcome-container">
//           {this.state.loading ? (
//             <CircularProgress />
//           ) : (
//               <div className="login-dialog">
//                 <img
//                   src={Logo}
//                   alt="app logo"
//                   className="app-login-logo"
//                 />
//                 <div>
//                   <div className="element" />
//                   <div className="login-text-fields">
//                     <TextField
//                       id="email"
//                       floatingLabelText="Email"
//                       fullWidth={true}
//                       onChange={this.onInputChange}
//                       onEnterKeyDown={() =>
//                         this.submitToAuth(this.props.onSignIn)
//                       }
//                       value={this.state.email}
//                       hintText="Email"
//                     />
//                     <TextField
//                       id="password"
//                       floatingLabelText="Password"
//                       fullWidth={true}
//                       onChange={this.onInputChange}
//                       onEnterKeyDown={() =>
//                         this.submitToAuth(this.props.onSignIn)
//                       }
//                       value={this.state.password}
//                       hintText="Password"
//                       type="password"
//                     />
//                   </div>
//                   <RaisedButton
//                     className="login-button"
//                     label="Log in"
//                     onTouchTap={() => this.submitToAuth(this.props.onSignIn)}
//                   />
//                   <RaisedButton
//                     className="login-button"
//                     label="Log in with Google"
//                     onTouchTap={() => oauthApi.signIn("signin")}
//                     icon={<GoogleIcon className="button-icon" />}
//                   />
//                   <div className="login-divider">
//                     <Divider />
//                   </div>
//                   <div className="login-text">Device, but no account?</div>
//                   <RaisedButton
//                     className="login-button"
//                     label="Register with Google"
//                     onTouchTap={() => oauthApi.signIn("register")}
//                     primary={true}
//                     icon={<GoogleIcon className="button-icon" />}
//                   />
//                 </div>
//               </div>
//             )}
//         </div>
//       </div>
//     );
//   }
// }
