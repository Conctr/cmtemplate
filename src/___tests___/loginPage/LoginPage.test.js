import React from 'react'
import { shallow  } from 'enzyme'
import LoginPage from '../../pages/LoginPage'
import toJson from 'enzyme-to-json'

describe('CommentList', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<LoginPage />)
  })
  it('hello world', () => {
    console.log('hello')
  })
})
