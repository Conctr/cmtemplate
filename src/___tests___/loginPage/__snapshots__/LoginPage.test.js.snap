// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentList matches snapshot 1`] = `undefined`;
